connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -1466,560 +1032,421 @@ public void testRestartConnectorAndTasksUnknownConnector() throws Exception {
         herder.restartTask(TASK0, callback);
         herder.tick();
 
-        try {
-            callback.get(1000L, TimeUnit.MILLISECONDS);
-            fail("Expected NotLeaderException to be raised");
-        } catch (ExecutionException e) {
-            assertTrue(e.getCause() instanceof NotAssignedException);
-            NotAssignedException notAssignedException = (NotAssignedException) e.getCause();
-            assertEquals(ownerUrl, notAssignedException.forwardUrl());
-        }
+        ExecutionException e = assertThrows(ExecutionException.class, () -> callback.get(1000, TimeUnit.MILLISECONDS));
+        assertEquals(NotAssignedException.class, e.getCause().getClass());
+        assertEquals(ownerUrl, ((NotAssignedException) e.getCause()).forwardUrl());
 
-        PowerMock.verifyAll();
+        verifyNoMoreInteractions(worker, member, configBackingStore, statusBackingStore);
     }
 
     @Test
-    public void testConnectorConfigAdded() {
+    public void testRequestProcessingOrder() {
+        final DistributedHerder.DistributedHerderRequest req1 = herder.addRequest(100, null, null);
+        final DistributedHerder.DistributedHerderRequest req2 = herder.addRequest(10, null, null);
+        final DistributedHerder.DistributedHerderRequest req3 = herder.addRequest(200, null, null);
+        final DistributedHerder.DistributedHerderRequest req4 = herder.addRequest(200, null, null);
+
+        assertEquals(req2, herder.requests.pollFirst()); // lowest delay
+        assertEquals(req1, herder.requests.pollFirst()); // next lowest delay
+        assertEquals(req3, herder.requests.pollFirst()); // same delay as req4, but added first
+        assertEquals(req4, herder.requests.pollFirst());
+    }
+
+    @Test
+    public void testConnectorConfigAdded() throws Exception {
         // If a connector was added, we need to rebalance
-        EasyMock.expect(member.memberId()).andStubReturn("member");
-        EasyMock.expect(member.currentProtocolVersion()).andStubReturn(CONNECT_PROTOCOL_V0);
+        when(member.memberId()).thenReturn("member");
+        when(member.currentProtocolVersion()).thenReturn(CONNECT_PROTOCOL_V0);
 
         // join, no configs so no need to catch up on config topic
         expectRebalance(-1, Collections.emptyList(), Collections.emptyList());
-        member.poll(EasyMock.anyInt());
-        PowerMock.expectLastCall();
+        doNothing().when(member).poll(anyLong());
+
+        herder.tick(); // join
 
-        // apply config
-        member.wakeup();
-        PowerMock.expectLastCall();
-        member.ensureActive();
-        PowerMock.expectLastCall();
         // Checks for config updates and starts rebalance
-        EasyMock.expect(configBackingStore.snapshot()).andReturn(SNAPSHOT);
-        member.requestRejoin();
-        PowerMock.expectLastCall();
+        doNothing().when(configBackingStore).refresh(anyLong(), any());
+        when(configBackingStore.snapshot()).thenReturn(SNAPSHOT);
```

Comment:
We have a utility method for this:
```suggestion
        expectConfigRefreshAndSnapshot(SNAPSHOT);
```

Reply:
Whoops 🤦